rclpy/src/rclpy/_rclpy.c

```diff
@@ -3583,6 +3595,21 @@ rclpy_convert_from_py_qos_policy(PyObject * Py_UNUSED(self), PyObject * args)
   qos_profile->depth = pyqos_depth;
   qos_profile->reliability = pyqos_reliability;
   qos_profile->durability = pyqos_durability;
+
+  rcl_duration_t * lifespan =
+    (rcl_duration_t *)PyCapsule_GetPointer(pyqos_lifespan, "rcl_duration_t");
+  qos_profile->lifespan = NANOSECONDS_TO_RMW_TIME(lifespan->nanoseconds);
```

Comment:
Need to check if lifespan is `NULL`. If so, this function must return `NULL` because an exception will be set.